### ISSUE TITLE
[android][expo]: statusBarStyle breaks translucent StatusBar

### ISSUE THREAD

**Issue by hirbod** [software-mansion/react-native-screens#1440]
## Description

I have 

```
  androidStatusBar: {
    translucent: true
  },
```

in my app.config.ts. This is working fine, but when I use `statusBarStyle` auto/light/dark, I get a white background on my Statusbar and it won't be translucent/floating over my content anymore.

## Steps To Reproduce

1. Make StatusBar translucent via app.config.ts
2. add `statusBarStyle` to any Screen 

### Expected behavior

Should not break the translucent/floating StatusBar

### Actual behavior

It is reverting the translucent StatusBar, thus breaking height calculations as well.

## Reproduction

```tsx
          <Navigator.Screen
            component={HomeStack}
            options={{
                statusBarStyle: 'inverted'
            }}
          />
```

## Platform

- [ ] iOS
- [x] Android
- [ ] Web
- [ ] Windows
- [ ] tvOS

## Workflow

- [X] Managed workflow (custom-dev-client)
- [x] Bare workflow (custom-dev-client)

## Package versions

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |  0.68.1       |
| @react-navigation/native               |    6.0.10    |
| @react-navigation/native-stack         |   6.6.2      |
| react-native-screens                   |  3.13.1       |
| react-native-safe-area-context         |   4.2.6      |
| react-native-gesture-handler           |  2.4.1       |
| react-native-reanimated                |   2.8.0      |
| expo                                   |  SDK 45 Beta-9       |

**Comment by a-eid** [comment]
actually this expo settings breaks the keyboard `adjustResize` somehow on android too.

**Comment by hirbod** [comment]
Yes it does indeed. I kinda semi fixed all of this issues together with https://github.com/kirillzyusko/react-native-keyboard-controller since it changes StatusBar and keyboard controller back to how it should be
